api/internal/handler/data_loader/route_export.go

```diff
@@ -0,0 +1,478 @@
+					}
+
+					if bsVal, ok := valueMap["body_schema"]; ok {
+						m := map[string]*openapi3.MediaType{}
+						reqBytes, _ := json.Marshal(&bsVal)
+						schema := &openapi3.Schema{}
+						err := json.Unmarshal(reqBytes, schema)
+						if err != nil {
+							log.Errorf("json marshal failed: %s", err)
+						}
+
+						m["*/*"] = &openapi3.MediaType{Schema: &openapi3.SchemaRef{Value: schema}}
+						requestBody.Content = m
+					}
+				}
+				continue
+			}
+			// analysis security plugins
+			securityEnv := &openapi3.SecurityRequirement{}
+			switch key {
+			case string(KeyAuth):
+				secSchemas["api_key"] = &openapi3.SecuritySchemeRef{Value: openapi3.NewCSRFSecurityScheme()}
+				securityEnv.Authenticate("api_key", " ")
+				secReq.With(*securityEnv)
+				continue
+			case string(BasicAuth):
+				secSchemas["basicAuth"] = &openapi3.SecuritySchemeRef{Value: &openapi3.SecurityScheme{
+					Type: "basicAuth",
+					Name: "basicAuth",
+					In:   "header",
+				}}
+				securityEnv.Authenticate("basicAuth", " ")
+				secReq.With(*securityEnv)
+				continue
+			case string(JWTAuth):
+				secSchemas["bearerAuth"] = &openapi3.SecuritySchemeRef{Value: openapi3.NewJWTSecurityScheme()}
+				securityEnv.Authenticate("bearerAuth", " ")
+				secReq.With(*securityEnv)
+				continue
+			}
+			plugins[key] = value
+		}
+		path.Security = secReq
+
+		if route.ServiceID != nil && servicePlugins != nil {
+			_servicePlugins, err := json.Marshal(servicePlugins)
+			if err != nil {
+				log.Errorf("MapToJson err: ", err)
+				return path, nil, nil, nil, err
+			}
+			_plugins, err := json.Marshal(plugins)
+			if err != nil {
+				log.Errorf("MapToJson err: ", err)
+				return path, nil, nil, nil, err
+			}
+			bytePlugins, err := utils.MergeJson(_servicePlugins, _plugins)
+			if err != nil {
+				log.Errorf("Plugins MergeJson err: ", err)
+				return path, nil, nil, nil, err
+			}
+			err = json.Unmarshal([]byte(bytePlugins), &plugins)
+			if err != nil {
+				log.Errorf("JsonToMapDemo err: ", err)
+				return path, nil, nil, nil, err
+			}
+		}
+	} else if route.Plugins == nil && route.ServiceID != nil {
+		plugins = servicePlugins
+	}
+	return path, secSchemas, paramsRefs, plugins, nil
+}
+
+// ParseRouteUris The URI and URIs of route are converted to paths URI in openapi3
+func ParseRouteUris(route *entity.Route, paths openapi3.Paths, paramsRefs []*openapi3.ParameterRef, pathItem *openapi3.PathItem) (openapi3.Paths, []*openapi3.ParameterRef) {
+	routeURIs := []string{}
+	if route.URI != "" {
+		routeURIs = append(routeURIs, route.URI)
+	}
+
+	if route.Uris != nil {
+		routeURIs = route.Uris
+	}
+
+	for _, uri := range routeURIs {
+		_time := time.Now().UnixNano() / 1e6
+		if strings.Contains(uri, "*") {
+			if _, ok := paths[strings.Split(uri, "*")[0]+"{params}"]; !ok {
+				paths[strings.Split(uri, "*")[0]+"{params}"] = pathItem
+			} else {
+				paths[strings.Split(uri, "*")[0]+"{params}"+"-APISIX-REPEAT-URI-"+strconv.FormatInt(_time, 10)] = pathItem
+			}
```

Comment:
This can be done in closure.

Reply:
Here I add a global variable, which is initialized to 0 every time I call the interface. In the loop, if it meets the conditions, it will automatically increase to 1. I don't know if this is OK

Reply:
done